src/controller/ai_controller.c

```diff
@@ -203,8 +203,9 @@ int char_to_act(int ch, int direction) {
  *
  * \return Void.
  */
-void chain_controller_cmd(controller *ctrl, int commands[], ctrl_event **ev) {
-    for(int i = 0; i < sizeof *commands; i++) {
+void chain_controller_cmd(controller *ctrl, const int commands[], ctrl_event **ev) {
+    size_t n = sizeof(*commands) / sizeof(int);
```

Comment:
I don't see how this can know the size of the array, does this really work?

Reply:
Oh yeah, I forgot to check. So this should technically work due to Compiler Magic, but I'm not sure if it works through function calls. I know that the old sizeof at least did not work correctly :/

Probably best make a test case.

Reply:
Or actually, probably best make a macro and pass a size_t size parameter.

Reply:
Kind of wonder if that ever worked, but definitely there is no way to know the size of that array from those parameters.

Reply:
It probably worked before because it was reading the pointer size, which was either 4 or 8. That would then have caused the loop to read all the content plus some extra :/

Reply:
I pushed something that should work to this branch, but also `chain_consider_tactics()` is broken the same way

Reply:
Added fixes for that.